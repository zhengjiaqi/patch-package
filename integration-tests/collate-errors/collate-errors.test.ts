import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "collate-errors",
  shouldProduceSnapshots: true,
})
