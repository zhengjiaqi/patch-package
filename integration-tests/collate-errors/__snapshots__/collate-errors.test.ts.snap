// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test collate-errors: left-pad, lodash, and zfs apply 1`] = `
"SNAPSHOT: left-pad, lodash, and zfs apply
patch-package 0.0.0
Applying patches...
left-pad@1.1.1 ✔
lodash@4.17.21 ✔
zfs@1.3.0 ✔
END SNAPSHOT"
`;

exports[`Test collate-errors: underscore does not apply, left-pad warns 1`] = `
"SNAPSHOT: underscore does not apply, left-pad warns

Warning: patch-package detected a patch file version mismatch

  Don't worry! This is probably fine. The patch was still applied
  successfully. Here's the deets:

  Patch file created for

    left-pad@1.1.1

  applied to

    left-pad@1.1.2
  
  At path
  
    node_modules/left-pad

  This warning is just to give you a heads-up. There is a small chance of
  breakage even though the patch was applied successfully. Make sure the package
  still behaves like you expect (you wrote tests, right?) and then run

    patch-package left-pad

  to update the version in the patch file name and make this warning go away.


**ERROR** Failed to apply patch for package underscore at path
  
    node_modules/underscore

  This error was caused because underscore has changed since you
  made the patch file for it. This introduced conflicts with your patch,
  just like a merge conflict in Git when separate incompatible changes are
  made to the same piece of code.

  Maybe this means your patch file is no longer necessary, in which case
  hooray! Just delete it!

  Otherwise, you need to generate a new patch file.

  To generate a new one, just repeat the steps you made to generate the first
  one.

  i.e. manually make the appropriate file changes, then run 

    patch-package underscore

  Info:
    Patch file: patches/underscore+1.0.3.patch
    Patch was made for version: 1.0.3
    Installed version: 1.12.0

---
patch-package finished with 1 warning(s), 1 error(s).
END SNAPSHOT"
`;
